src/App.tsx

```diff
@@ -15,48 +16,110 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  Default,
+  Alphabetical,
+  Length,
+}
+
 export const App: React.FC = () => {
+  const originalGoods = [...goodsFromServer];
+  const [visibleGoods, setVisibleGoods] = useState<string[]>(originalGoods);
+  const [sortType, setSortType] = useState<SortType>(SortType.Default);
+  const [isReverseActive, setIsReverseActive] = useState(false);
+
+  const sortGoods = () => {
+    let sortedGoods: string[] = [...visibleGoods];
+
+    switch (sortType) {
+      case SortType.Alphabetical:
+        sortedGoods = sortedGoods.sort();
+        break;
+      case SortType.Length:
+        sortedGoods = sortedGoods.sort(
+          (good1, good2) => good1.length - good2.length,
+        );
+        break;
+      default:
+        break;
+    }
+
+    if (isReverseActive) {
+      sortedGoods.reverse();
+    }
+
+    setVisibleGoods(sortedGoods);
+  };
```

Comment:
move this sorting function outside App component. Function should return a list of goods